src/finufft_core.cpp

```diff
@@ -978,68 +969,101 @@ template int FINUFFT_PLAN_T<double>::setpts(BIGINT nj, double *xj, double *yj, d
 
 // EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE
 template<typename TF>
-int FINUFFT_PLAN_T<TF>::execute(std::complex<TF> *cj, std::complex<TF> *fk) {
+int FINUFFT_PLAN_T<TF>::execute_internal(TC *cj, TC *fk, bool adjoint, int ntrans_actual,
+                                         TC *aligned_scratch, size_t scratch_size) const {
   /* See ../docs/cguru.doc for current documentation.
 
    For given (stack of) weights cj or coefficients fk, performs NUFFTs with
    existing (sorted) NU pts and existing plan.
-   For type 1 and 3: cj is input, fk is output.
-   For type 2: fk is input, cj is output.
+   For adjoint == false:
+     For type 1 and 3: cj is input, fk is output.
+     For type 2: fk is input, cj is output.
+   For adjoint == true:
+     For type 1 and 3: fk is input, cj is output.
+     For type 2: cj is input, fk is output.
    Performs spread/interp, pre/post deconvolve, and FFT as appropriate
    for each of the 3 types.
    For cases of ntrans>1, performs work in blocks of size up to batchSize.
    Return value 0 (no error diagnosis yet).
    Barnett 5/20/20, based on Malleo 2019.
+
+   Additional parameters introducd by MR, 02/2025
+
+   adjoint: if false, behave as before; if true, compute the adjoint of the
```

Comment:
see my overall PR review discussion

Reply:
I'm very happy to adjust the interface in any way that makes it easier for finufft users to get used to.  This PR is (at the moment) mostly a demonstration of how things can be done. How they are presented is always up to the Finufft team.